### PR TITLE
2021-02-19-Kubernetes-Portability-Reimagined

### PR DESCRIPTION
**Pull request by bhkamath**
At the time of writing this article, I envisioned it to be a blog.  However, if you find it appropriate this may also add value to descriptions and examples on portability, ConfigMap and Secrets.

### CODE REVIEW

**Comment by bhkamath** [comment]
I have successfully signed CNCF Individual Contributor License Agreement
Screenshot 2021-02-20 at 11 01 39 AM

**Comment by celestehorgan** [comment]
Hi @bhkamath,

1) Blog posts need certain metadata (frontmatter) at the top to generate things like title and post dates. See [any other blog post](https://github.com/kubernetes/website/blob/master/content/en/blog/_posts/2015-04-00-Introducing-Kubernetes-V1Beta3.md) currently in the repository as a nexample.
2) Blog posts also follow a certain file format, `YYYY-MM-DD-Title-No-Spaces.md`. Please rename your file in the appropriate manner. 

Once you do that we can begin reviewing this contribution.

**Review by bhkamath — COMMENTED** [review]
I have made an attempt to rectify and apply the pattern being followed.  The meta data is in text format since graphical content is not permitted.  Also, I have an image and call-outs associated with the YAML file if permitted.  Please do let me know if this is appropriate.

**Review by bhkamath — COMMENTED** [review]
Hi,

I have adhered to the filenaming convention and front matter contents.  Request you to review the same.

Thanks,

File: content/en/blog/_posts/2021-02-19-Kubernetes-Portability-Reimagined [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+Environment variables are likely to differ between deployments such as test and production.  These configuration parameters may be independently managed and scaled with the help of ConfigMap and Secrets resources, thus achieving separation of code from configuration data.  While Secrets provides basic security, base64 encoding; the ConfigMap and Secrets data is made available to nodes and Pods on need basis for the lifecycle of the Pod but not otherwise, although declared in the cluster.
+
+In the following example, we present a Pod, ConfigMap and Secrets that decouples the environment variables - counter, sleep-interval, username and password from embedding into a Container.  To refer these environment variables in a container of a Pod, one needs to perform the following steps
+
+Declare key-value pairs into a Kubernetes namespace using ConfigMap and Secrets object
+Refer to these key-value pairs as data volume files or environment variables in a Container of a Pod
+
+$ cat kube-configmap.yaml
+Declaration of 2
+Key-value pairs
```
Comment (sftim) [inline content/en/blog/_posts/2021-02-19-Kubernetes-Portability-Reimagined:24]: Hi. Can you put this inside a [code fence](https://www.markdownguide.org/extended-syntax/#fenced-code-blocks)? Also, do the same for other blocks of code in the page.

File: content/en/blog/_posts/2021-02-19-Kubernetes-Portability-Reimagined [hunk 2]
```diff
@@ -0,0 +1,135 @@
+---
+layout: blog
+title: Kubernetes Portability Reimagined
+date:  2021-02-19
+slug:  Kubernetes-Portability-Reimagined
+url:   /blog/2021/02/Kubernetes-Portability-Reimagined
```
Comment (sftim) [inline content/en/blog/_posts/2021-02-19-Kubernetes-Portability-Reimagined:6]: This should be:
```suggestion
slug:  kubernetes-portability-reimagined
```

**Review by sftim — COMMENTED** [review]
Hi @bhkamath 

I'm not sure that this article as it stands is right for our target audience: people reading the Kubernetes blog often already know what Kubernetes is.

Maybe you can add some more text to explain good practice for [separating configuration from code,](https://12factor.net/config) and how to do this specifically with Kubernetes?

I think if you took the explanation further and tidied the article up, we could look at including it.

**Comment by sftim** [comment]
> graphical content is not permitted

Images are OK in the blog. Please use SVG for diagrams.

**Comment by reylejano** [comment]
Hi @bhkamath , 

I believe there are several things that should be addressed with this blog:
- white space; use one space between words and after a period to end a sentence. There are a few instances of two spaces between words
- [use UpperCamelCase for API objects](https://kubernetes.io/docs/contribute/style/style-guide/) e.g. ReplicaSet not Replicaset
- 80 character limit for the width; this makes it easier to review
- proper code blocks; please look at previous blog or documentation examples
- the items in the References section should be linked
- Use headings to separate sections

**Comment by reylejano** [comment]
I believe the context of this blog is covered in the current documentation in the [concepts](https://kubernetes.io/docs/concepts/) and [tasks](https://kubernetes.io/docs/tasks/) sections.
@bhkamath , what additional value does the blog deliver to the audience?

**Comment by sftim** [comment]
I'll close this @bhkamath 

We'd be happy to take a look at other proposed articles, and you're also very welcome to work on the Kubernetes documentation.
/close

**Pull request closed without merging**